stories/rank-step.stories.js

```diff
@@ -0,0 +1,296 @@
+  args: {
+    pointRankGroupList: mergeRanksIntoReviewPoints(rankPoints, [rank1preMost, rank2preNeutral, rank3preLeast]),
+    discussionId,
+    round,
+  },
+}
+
+export const PartialWithInitialRank = {
+  args: {
+    pointRankGroupList: mergeRanksIntoReviewPoints(rankPoints, [rank1preMost, rank2preNeutral]),
+    discussionId,
+    round,
+  },
+}
+
+export const onDoneIsCalledIfInitialData = {
+  args: {
+    pointRankGroupList: mergeRanksIntoReviewPoints(rankPoints, [rank1preMost]),
+    discussionId,
+    round,
+  },
+  play: async ({ canvasElement }) => {
+    const canvas = within(canvasElement)
+    await waitFor(() => {
+      expect(onDoneResult(canvas)).toMatchObject({
+        count: 1,
+        onDoneResult: {
+          valid: false,
+          value: 0.3333333333333333,
+        },
+      })
+    })
+  },
+}
+
+export const onDoneIsCalledAfterUserChangesRank = {
+  args: {
+    pointRankGroupList: mergeRanksIntoReviewPoints(rankPoints, [rank1preMost]),
+    discussionId,
+    round,
+  },
+  play: async ({ canvasElement }) => {
+    const canvas = within(canvasElement)
+    const categories = canvas.getAllByText('Neutral')
+    await userEvent.click(categories[0])
+
+    await waitFor(() => {
+      expect(onDoneResult(canvas)).toMatchObject({
+        count: 2,
+        onDoneResult: {
+          valid: false,
+          value: 0.3333333333333333,
+          delta: { ...rank1preMost, category: 'neutral' },
+        },
+      })
+    })
+  },
+}
+
+// need to have one source of data {rankPoints}, but split it into defaultValue representing what should be setup in the context before RankStep renders
+// and topWhyById and postRankByParentId which are for the simulated response to the get-user-post-ranks-and-top-ranked-whys socket api call
+function getRankArgsFrom(rankPoints) {
+  const cn = rankPointsToContext(rankPoints)
+  const { preRankByParentId, ...defaultValue } = { ...cn, round, discussionId }
+  return { preRankByParentId, defaultValue }
+}
+
+// sets up the socket api mocks and renders the component
+const rankStepTemplate = args => {
+  // topWhyById and postRankByParentId are taken from args to uses by the api call
+  const { preRankByParentId, ...otherArgs } = args
+  useState(() => {
+    // execute this code once, before the component is initally rendered
+    // the api call will provide the new data for this step
+    window.socket._socketEmitHandlers['get-user-ranks'] = (discussionId, round, ids, cb) => {
+      window.socket._socketEmitHandlerResults['get-user-ranks'] = [discussionId, round, ids]
+      setTimeout(() => {
+        const ranks = Object.values(pprRankByParentId) // back to array
+        cb([ranks])
+      })
+    }
+    window.socket._socketEmitHandlers['upsert-rank'] = (rank, cb) => {
+      window.socket._socketEmitHandlerResults['upsert-rank'] = rank
+      cb && cb()
+    }
+  })
+  return <RankStep {...otherArgs} />
+}
+
+export const rankStepWithPartialDataAndUserUpdate = {
```

Comment:
This test, http://localhost:6006/?path=/story/rank-step--rank-step-with-partial-data-and-user-update and the next one 
http://localhost:6006/?path=/story/rank-step--rank-step-with-top-down-update are failing the Interaction tests.  

One problem I see is that line 187 says pprRankByParentId instead of preRankByParentId

After the test runs, some of the items should have ranks, but currently none do.